New incremental mode (-i) crasher

I've seen this a few times myself and now it's being reported. The traceback ends with
```
  File "/Users/x/src/server/.mypy/venv/lib/python3.6/site-packages/mypy/types.py", line 388, in accept
    return visitor.visit_instance(self)
  File "/Users/x/src/server/.mypy/venv/lib/python3.6/site-packages/mypy/fixup.py", line 153, in visit_instance
    node = lookup_qualified(self.modules, type_ref, self.quick_and_dirty)
  File "/Users/x/src/server/.mypy/venv/lib/python3.6/site-packages/mypy/fixup.py", line 237, in lookup_qualified
    stnode = lookup_qualified_stnode(modules, name, quick_and_dirty)
  File "/Users/x/src/server/.mypy/venv/lib/python3.6/site-packages/mypy/fixup.py", line 252, in lookup_qualified_stnode
    assert '.' in head, "Cannot find %s" % (name,)
AssertionError: Cannot find _importlib_modulespec.ModuleType
```

--- a/mypy/fixup.py
+++ b/mypy/fixup.py
@@ -243,7 +243,9 @@ def lookup_qualified(modules: Dict[str, MypyFile], name: str,
 
 def lookup_qualified_stnode(modules: Dict[str, MypyFile], name: str,
                             quick_and_dirty: bool) -> Optional[SymbolTableNode]:
-    name = rev_module_rename_map.get(name, name)
+    if '_importlib_modulespec' in modules:
+        # we are using python 3, so renaming is necessary
+        name = rev_module_rename_map.get(name, name)
     head = name
     rest = []
     while True:

--- a/mypy/semanal.py
+++ b/mypy/semanal.py
@@ -117,13 +117,18 @@ obsolete_name_mapping = {
     'typing.typevar': 'typing.TypeVar',
 }
 
+# Used for python 3 only
 # Rename objects placed in _importlib_modulespec due to circular imports
+# Safe in python 2 because no one will ask for these keys
 module_rename_map = {
     '_importlib_modulespec.ModuleType': 'types.ModuleType',
     '_importlib_modulespec.ModuleSpec': 'importlib.machinery.ModuleSpec',
     '_importlib_modulespec.Loader': 'importlib.abc.Loader'
 }
 
+# Must not be used in python 2 (it will replace correct names with incorrect)
+# The check for version is currently made at the point of use
+# TODO: consider moving these maps from global scope to somewhere where python version is known
 rev_module_rename_map = {v: k for (k, v) in module_rename_map.items()}
 
 # Hard coded type promotions (shared between all Python versions).

